homeassistant/components/home_connect/sensor.py

```diff
@@ -38,47 +42,205 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+EVENT_OPTIONS_TRANSLATION_KEYS = {
+    BSH_EVENT_PRESENT_STATE_PRESENT: "present",
+    BSH_EVENT_PRESENT_STATE_CONFIRMED: "confirmed",
+    BSH_EVENT_PRESENT_STATE_OFF: "off",
+}
+
+
 @dataclass(frozen=True, kw_only=True)
 class HomeConnectSensorEntityDescription(SensorEntityDescription):
     """Entity Description class for sensors."""
 
-    device_class: SensorDeviceClass | None = SensorDeviceClass.ENUM
-    options: list[str] | None = field(
-        default_factory=lambda: ["confirmed", "off", "present"]
-    )
-    desc: str
-    appliance_types: tuple[str, ...]
+    options_translation_keys: dict[str, str] | None = None
+    default_value: str | None = None
+    appliance_types: tuple[str, ...] | None = None
+    sign: int = 1
+
 
+BSH_PROGRAM_SENSORS = (
+    HomeConnectSensorEntityDescription(
+        key="BSH.Common.Option.RemainingProgramTime",
+        device_class=SensorDeviceClass.TIMESTAMP,
```

Comment:
Is this timestamp or duration?

Reply:
It was timestamp before:
https://github.com/home-assistant/core/blob/0d795aad16fa095f93f305e428ec26d7dadf93e6/homeassistant/components/home_connect/api.py#L192-L198

Reply:
It looks like the value that comes from the API are seconds, but then is converted to a timestamp equal to now + seconds retrieved from API:
https://github.com/home-assistant/core/blob/36b4db26c30ff335a673af53d680ea1e3fbffda7/homeassistant/components/home_connect/sensor.py#L308-L336

That makes me think that maybe we should do at least one option of the following:

- Change translation key to "Finalization date" or similar
- Expose the remaining seconds

Doing both options (so another Entity description will be needed) will cause issues with unique IDs because both will use the same BSH key...

I would chose the second option, and if the user wants to have the timestamp, a template entity can be use to achieve that

WDYT?

Reply:
Ideally we display the api data as raw as possible, so I'd opt for deprecating the current one and adding one that is actually the real value

Reply:
Got it!
I will create a PR doing that

Reply:
This lead to state changes be every second. We usually prefer timestamps

Reply:
Oh that is actually a good point, haven't thought about that one

Reply:
Should we rename it then to fit the actual functionality?
Do you have any suggestion?

Reply:
Sounds good